### ISSUE TITLE
False positive for empty_xctest_method when assigning statements to _

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#3691]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I'm getting a false positive for `empty_xctest_method` with code like the following:

```
    func testStuff() {
        _ = 4 + 4
    }
```

and 

```
    func testStuff() {
        _ = 4 + 4
        _ = 4 + 4
    }
```

But not

```
    func testStuff() {
        let aaa = 4 + 4
    }
```

or 

```
    func testStuff() {
        _ = 4 + 4
        let aaa = 4 + 4
    }
```

When these methods are members of a subclass of XCTestCase

Sample input:

```
import XCTest

class MyTests: XCTestCase {
    func testStuff() {
        _ = 4 + 4
    }
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --config swiftlint.yml  t.swift

Linting Swift files at paths t.swift
Linting 't.swift' (1/1)
/tmp/t.swift:4:5: warning: Empty XCTest Method Violation: Empty XCTest method should be avoided. (empty_xctest_method)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version - 0.43.1
* Installation method used - HomeBrew
* Paste your configuration file:

```yml
opt_in_rules:
  - empty_xctest_method
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?

No

* Which Xcode version are you using (check `xcodebuild -version`)?

```
Xcode 12.4
Build version 12D4e
```

* Do you have a sample that shows the issue?

```swift
import XCTest

class MyTests: XCTestCase {
    func testStuff() {
        _ = 4 + 4
    }
}
```

**Comment by KaneCheshire** [comment]
I reported a similar (but different) issue with this rule here too a couple of months ago: https://github.com/realm/SwiftLint/issues/3647

**Comment by KaneCheshire** [comment]
This looks like an issue with SourceKitten so I've opened an issue there and referenced this issue plus my other issue
